### PR TITLE
Merge session replay branch into sdkv2

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: features/dd-sdk-android-session-replay/api/apiSurface [hunk 1]
```diff
@@ -22,12 +22,26 @@ interface com.datadog.android.sessionreplay.internal.recorder.OptionSelectorDete
...
-abstract class com.datadog.android.sessionreplay.internal.recorder.mapper.BaseWireframeMapper<T: android.view.View, S: com.datadog.android.sessionreplay.model.MobileSegment.Wireframe> : WireframeMapper<T, S>
-  constructor(com.datadog.android.sessionreplay.utils.StringUtils = StringUtils, com.datadog.android.sessionreplay.utils.ViewUtils = ViewUtils)
+class com.datadog.android.sessionreplay.internal.recorder.base64.Base64Serializer
+interface com.datadog.android.sessionreplay.internal.recorder.base64.ImageCompression
+  fun getMimeType(): String?
+  fun compressBitmapToStream(android.graphics.Bitmap): java.io.ByteArrayOutputStream
+class com.datadog.android.sessionreplay.internal.recorder.base64.WebPImageCompression : ImageCompression
+  override fun getMimeType(): String?
+  override fun compressBitmapToStream(android.graphics.Bitmap): java.io.ByteArrayOutputStream
+  companion object
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay/api/apiSurface:32]: probably shouldn't be here, but it is not important

File: features/dd-sdk-android-session-replay/api/apiSurface [hunk 2]
```diff
@@ -22,12 +22,26 @@ interface com.datadog.android.sessionreplay.internal.recorder.OptionSelectorDete
...
   protected fun resolveViewId(android.view.View): Long
   protected fun colorAndAlphaAsStringHexa(Int, Int): String
   protected fun resolveViewGlobalBounds(android.view.View, Float): com.datadog.android.sessionreplay.internal.recorder.GlobalBounds
   protected fun android.graphics.drawable.Drawable.resolveShapeStyleAndBorder(Float): Pair<com.datadog.android.sessionreplay.model.MobileSegment.ShapeStyle?, com.datadog.android.sessionreplay.model.MobileSegment.ShapeBorder?>?
+  protected fun resolveChildDrawableUniqueIdentifier(android.view.View): Long?
+  protected fun getWebPMimeType(): String?
+  protected fun handleBitmap(android.util.DisplayMetrics, android.graphics.drawable.Drawable, com.datadog.android.sessionreplay.model.MobileSegment.Wireframe.ImageWireframe)
+  override fun startProcessingImage()
+  override fun finishProcessingImage()
+  companion object
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay/api/apiSurface:44]: probably shouldn't be here, but it is not important

**Review by 0xnm — APPROVED** [review]

File: features/dd-sdk-android-session-replay/api/apiSurface (hunk 1 above)
Comment (mariusc83) [reply features/dd-sdk-android-session-replay/api/apiSurface:32]: it is, it is exposed by the `BaseWireframeMapper`. We took the decision to push it now and see how we can hide this better later. Not very important for now.

File: features/dd-sdk-android-session-replay/api/apiSurface (hunk 2 above)
Comment (mariusc83) [reply features/dd-sdk-android-session-replay/api/apiSurface:44]: same as above

**Pull request merged**